include/range/v3/view/subrange.hpp

```diff
@@ -0,0 +1,431 @@
+#endif
+                return *this;
+            }
+        };
+
+        template<typename I, typename S, subrange_kind K>
+        struct is_referenceable_range<subrange<I, S, K>> : std::true_type {};
+
+        template<typename I, typename S = I, CONCEPT_REQUIRES_(Sentinel<S, I>())>
+        using sized_subrange = subrange<I, S, subrange_kind::sized>;
+
+#if RANGES_CXX_DEDUCTION_GUIDES >= RANGES_CXX_DEDUCTION_GUIDES_17
+        template<typename I, typename S, CONCEPT_REQUIRES_(Sentinel<S, I>())>
+        subrange(I, S, difference_type_t<I>) -> subrange<I, S, subrange_kind::sized>;
+
+        template<typename P, CONCEPT_REQUIRES_(detail::IteratorSentinelPair<P>())>
+        subrange(P) -> subrange<
+            meta::_t<std::tuple_element<0, P>>,
+            meta::_t<std::tuple_element<1, P>>>;
+
+        template<typename P, CONCEPT_REQUIRES_(detail::IteratorSentinelPair<P>())>
+        subrange(P, difference_type_t<meta::_t<std::tuple_element<0, P>>>) ->
+            subrange<meta::_t<std::tuple_element<0, P>>,
+                meta::_t<std::tuple_element<1, P>>, subrange_kind::sized>;
+
+        template<typename R, CONCEPT_REQUIRES_(ReferenceableRange<R>())>
+        subrange(R &&) -> subrange<iterator_t<R>, sentinel_t<R>,
+            static_cast<subrange_kind>(SizedRange<R>() ||
+                SizedSentinel<sentinel_t<R>, iterator_t<R>>())>;
+
+        template<typename R, CONCEPT_REQUIRES_(ReferenceableRange<R>())>
+        subrange(R &&, range_difference_type_t<R>) ->
+            subrange<iterator_t<R>, sentinel_t<R>, subrange_kind::sized>;
+#endif // RANGES_CXX_DEDUCTION_GUIDES
+
+        template<typename R, CONCEPT_REQUIRES_(Range<R>())>
+        using safe_subrange_t =	meta::if_<
+            ReferenceableRange<R>,
+            subrange<iterator_t<R>>,
+            dangling<subrange<iterator_t<R>>>>;
+
+#if RANGES_CXX_IF_CONSTEXPR >= RANGES_CXX_IF_CONSTEXPR_17 && \
+    RANGES_CXX_RETURN_TYPE_DEDUCTION >= RANGES_CXX_RETURN_TYPE_DEDUCTION_14 && \
+    RANGES_CXX_CONSTEXPR >= RANGES_CXX_CONSTEXPR_14
+        template<std::size_t N, typename I, typename S, subrange_kind K,
+            CONCEPT_REQUIRES_(N < 2)>
+        constexpr auto get(subrange<I, S, K> const &r) {
+            if constexpr (N == 0) {
+                return r.begin();
+            } else {
+                return r.end();
+            }
+        }
+#else
+        template<std::size_t N, typename I, typename S, subrange_kind K,
+            CONCEPT_REQUIRES_(N == 0)>
+        constexpr I get(subrange<I, S, K> const &r) {
+            return r.begin();
+        }
+
+        template<std::size_t N, typename I, typename S, subrange_kind K,
+            CONCEPT_REQUIRES_(N == 1)>
+        constexpr S get(subrange<I, S, K> const &r) {
+            return r.end();
+        }
+#endif
+
+        template<typename I, typename S, CONCEPT_REQUIRES_(Sentinel<S, I>())>
+        constexpr auto make_subrange(I i, S s)
```

Comment:
Should this be named `view::subrange`?

Reply:
I'd rather leave it as `ranges::make_subrange`.